rbc/irtools.py

```diff
@@ -36,7 +38,13 @@ def get_called_functions(library, funcname=None):
     for block in func.blocks:
         for instruction in block.instructions:
             if instruction.opcode == 'call':
-                name = list(instruction.operands)[-1].name
+                instr = list(instruction.operands)[-1]
+                name = instr.name
+                # ignore in a call to a function pointer
+                if 'bitcast' in str(instr):
+                    continue
+                if name == '':
+                    continue
```

Comment:
In the LLVM C++ API, this is usually done by checking if `CallInst->getCalledFunction() == nullptr || CallInst->getCalledFunction()->isDeclaration()`, does `llvmlite` not have similar methods?

Reply:
Unfortunately, no. The available methods for a ValueRef are:

```python
(Pdb++) print(instruction)
  %.32 = call i8* bitcast (%Struct.MemInfo.5* (i64, i8*)* @NRT_MemInfo_new_varsize_dtor to i8* (i64, i8*)*)(i64 64, i8* bitcast (void (i8*)* @.dtor.list.int64 to i8*))
(Pdb++) instruction.has_name
*** AttributeError: 'ValueRef' object has no attribute 'has_name'
(Pdb++) vars(instruction)
{'_kind': 'instruction', '_parents': {'module': <llvmlite.binding.module.ModuleRef object at 0x7feb8bf0bf10>, 'function': <llvmlite.binding.value.ValueRef object at 0x7feb8bf0bfa0>, 'block': <llvmlite.binding.value.ValueRef object at 0x7feb8b62e7a0>}, '_ptr': <llvmlite.binding.ffi.LP_LLVMValue object at 0x7feb8b832140>, '_as_parameter_': <llvmlite.binding.ffi.LP_LLVMValue object at 0x7feb8b832140>, '_capi': <llvmlite.binding.ffi._lib_wrapper object at 0x7fec59bedc00>}
(Pdb++) dir(instruction)
['__bool__', '__class__', '__del__', '__delattr__', '__dict__', '__dir__', '__doc__', '__enter__', '__eq__', '__exit__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__nonzero__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_as_parameter_', '_capi', '_closed', '_dispose', '_kind', '_owned', '_parents', '_ptr', 'add_function_attribute', 'arguments', 'attributes', 'block', 'blocks', 'close', 'closed', 'detach', 'function', 'instruction', 'instructions', 'is_argument', 'is_block', 'is_declaration', 'is_function', 'is_global', 'is_instruction', 'is_operand', 'linkage', 'module', 'name', 'opcode', 'operands', 'storage_class', 'type', 'visibility']
```

Reply:
Bummer :(